### ISSUE TITLE
Incorrect converter used when multiple field maps exist for the same a and b type.

### ISSUE THREAD

**Issue by brittinfontenot** [orika-mapper/orika#186]
When setting up a class map I set up two field maps, both of which map from String to String. Each field map uses a converter that trims the string down to a specific length. The problem is that in this scenario the first converter registered to a field map is used by both field maps and the second converter is ignored. Any additional converters registered to other String to String field maps are ignored as well.

https://gist.github.com/brittinfontenot/e70fe3ad92edebb818500c7286226aef

**Comment by frami** [comment]
I have the same issue with the version 1.5.0

**Comment by brabenetz** [comment]
With @5f6a736 the CustomConverter got a equals and hashcode method.
I think this should be removed from CustomConverter (Maybe the Builtin*Converters should implement them)
In the meanwhile you can try to implement your own equals & hashCode in your Converters.

I also saw that the BuiltinBidirectionalConverter implements the equals method without hashcode which should be fixed.

**Comment by brabenetz** [comment]
I would prefer to add apache commons lang as dependency, so the EqulasBuilder could be used in Orika:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html
Maybe for 1.6 ?

**Comment by frami** [comment]
You are right. I implemented now my own equals and hashcode method. Now it works as expected.
Thank you.

**Comment by brittinfontenot** [comment]
Thanks @brabenetz. Adding equals and hashcode to my converter works.

**Comment by nathansgreen** [comment]
If you add a commons lang dependency, can it be optional?

**Comment by brabenetz** [comment]
If I add commons lang3, then it will not be optional.
But I'm curious, what could be the problem with apache commons lang3 library?

**Comment by nathansgreen** [comment]
Some of my projects use only Guava ([similar functionality](https://gist.github.com/rocketraman/653af25ee3bf72ca497f)) and exclude a lot of commons stuff. Any time a new dependency pops up it has to be looked at.

**Comment by grubeninspekteur** [comment]
Any updates on whether removing equals and hashCode from CustomConverter is planned? I assume this was done in good faith to cache converters, but it's not documented and very confusing. I could raise a PR.
